src/pages/todo/sagas.js

```diff
@@ -3,13 +3,18 @@ import { fork } from 'redux-saga/effects'
 
 import {
   LIST_TODOS,
-  VIEW_TODO
+  VIEW_TODO,
+  CREATE_TODO,
+  UPDATE_TODO,
+  DELETE_TODO,
+  MOVE_TODO
 } from './actions';
 
 export default function* todoSagas() {
   yield [
     fork(watchApi, LIST_TODOS, '/v1/todos'),
-    fork(watchApi, VIEW_TODO, '/v1/todos/${id}')
+    fork(watchApi, VIEW_TODO, '/v1/todos/${id}'),
```

Comment:
I rethink that
`fork(watchApi, CREATE_TODO, '/v1/todos'),`
should be `fork(watchApi, CREATE_TODO, '/v1/todos', 'post'),`

and then in common/api.js `export function* watchApi(requestTypes, path)` should be 
```
export function* watchApi(requestTypes, path, method='get') {
  yield* takeLatest(requestTypes.REQUEST, fetchApi, requestTypes, path, method)
}
```

and `export function* fetchApi(requestTypes, path, action)` should be 
```
export function* fetchApi(requestTypes, path, action, method) {
  try {
    path = format(path, action)
    const options = getFetchOptions(method, action);
    const data = yield call(callApi, path, options);
    yield put({type: requestTypes.SUCCESS, data});
  } catch (error) {
    yield put({type: requestTypes.FAILURE, error});
  }
}
```

and `function optionFilter(data)` should be 
```
function getFetchOptions(method, action) {
  let headers = Object.assign({'Accept': 'application/json'}, action.headers)
  let data = action.data
  if (!action.raw) {
    headers = Object.assign(headers, {'Content-Type': 'application/json'})
    data = action.data ? JSON.stringify(action.data) : null
  }
  return {
    method: method,
    headers: headers,
    body: data
  }
}
```